drivers/hmis/spec/models/hmis/hud/enrollment_spec.rb

```diff
@@ -227,4 +227,26 @@
       end
     end
   end
+
+  describe 'enrollment data collection features' do
+    # see the system test drivers/hmis/spec/system/hmis/data_collection_features_spec.rb for more comprehensive tests
+
+    context 'when there is a more specific rule that doesnt match the enrollment' do
+      let!(:p1) { create :hmis_hud_project, data_source: ds1, organization: o1, project_type: 4 }
+      let!(:definition) { create :hmis_form_definition, title: 'CLS', role: 'CURRENT_LIVING_SITUATION', identifier: 'custom_cls' }
+      # Rule 1 is more specific for this project, but doesn't apply to the specific enrollment in question
+      let!(:rule1) { create :hmis_form_instance, entity: p1, data_collected_about: 'HOH', role: 'CURRENT_LIVING_SITUATION', definition_identifier: 'custom_cls' }
+      # Rule 2 is less specific for the project, but applies to the non-HOH enrollment in question
+      let!(:rule2) { create :hmis_form_instance, entity: nil, project_type: p1.project_type, role: 'CURRENT_LIVING_SITUATION', definition_identifier: 'custom_cls' }
+
+      let!(:hoh) { create :hmis_hud_client, first_name: 'Annie', data_source: ds1 }
+      let!(:spouse) { create :hmis_hud_client, first_name: 'Jessie', data_source: ds1 }
+      let!(:hoh_enrollment) { create :hmis_hud_enrollment, data_source: ds1, project: p1, client: hoh, entry_date: 1.month.ago, household_id: 'household1', relationship_to_hoh: 1 }
+      let!(:spouse_enrollment) { create :hmis_hud_enrollment, data_source: ds1, project: p1, client: spouse, entry_date: 1.month.ago, household_id: 'household1', relationship_to_hoh: 3 }
```

Comment:
Just so you know, you don't _need_ to create the clients explicitly. If there aren't specific attributes that you need on the client, you can allow the enrollment factory to create them:
```suggestion
      let!(:hoh_enrollment) { create :hmis_hud_enrollment, data_source: ds1, project: p1, entry_date: 1.month.ago, household_id: 'household1', relationship_to_hoh: 1 }
      let!(:spouse_enrollment) { create :hmis_hud_enrollment, data_source: ds1, project: p1, entry_date: 1.month.ago, household_id: 'household1', relationship_to_hoh: 3 }
```

Not a necessary change here, just pointing it out. In this case it is probably convenient that each client is assigned a distinct `first_name` so we can easily tell them apart when debugging.

Reply:
Oops - I thought I was adding that comment to the system test (hence the debugging comment). But same goes here.

Reply:
Cool, yeah I think here in the unit test it makes sense to do this. I'll keep the system test as is for convenience in debugging.